lib/lint/index.js

```diff
@@ -91,7 +94,11 @@ function lint(config){
     l10nInfo: true
   }).concat([
     function(flow){
-      flow.result = require('./reporter/process-warns')(flow.warns, options.filter);
+      if (options.warnUnusedFiles)
+        collectUsed.files(flow);
```

Comment:
Bad pattern: function has a hidden side effect. As I realised it adds `usedFiles` to `flow` – it's confusing and should be explicit.